3_extracting_features/DPutils.py

```diff
@@ -0,0 +1,237 @@
+            # be in this specific naming structure
+            frame_segmentations = frame_segmentations.rename(
+                columns={
+                    "Location_Center_X": "Nuclei_Location_Center_X",
+                    "Location_Center_Y": "Nuclei_Location_Center_Y",
+                }
+            )
+
+            locations_save_path.parents[0].mkdir(parents=True, exist_ok=True)
+            frame_segmentations.to_csv(locations_save_path, index=False)
+
+
+def compile_project(
+    project_path: pathlib.Path,
+    checkpoint_name: str,
+    annotations_path: pathlib.Path,
+    images_load_path: pathlib.Path,
+    segmentation_data_path: pathlib.Path,
+    object: str,
+):
+    """Compile DP project for specified object of interest.
+    Args:
+        project_path (pathlib.Path): Path to compile DP project to for an object
+        checkpoint_name (str): Name of checkpoint to use in DP project (must be located in DP_files/)
+        annotations_path (pathlib.Path): Path to NF1 annotations data
+        images_load_path (pathlib.Path): Path to load NF1 illumination corrected images to extract features from
+        segmentation_data_path (pathlib.Path): Path to folder with segmentation tsv files 
+        object (str): Object to compile project for, must be "nuc" or "cyto"
+    """
+    # Make project directory
+    project_path.mkdir(parents=True, exist_ok=True)
+
+    # Copy necessary DP files from DP_files/ to DP project
+    config_name = f"NF1_{object}_config.json"
+    copy_DP_files(project_path, config_name, checkpoint_name)
+
+    # Compile and save object index.csv file to DP project for object
+    # Read in annotations file to use in function
+    annotations = pd.read_csv(annotations_path)
+
+    # Create path for index.csv file within the project path
+    index_save_path = pathlib.Path(f"{project_path}/inputs/metadata/index.csv")
+    index_save_path.parents[0].mkdir(parents=True, exist_ok=True)
+    print("compiling index.csv file...")
+
+    # Create path for the images in the DP project
+    DP_images_path = pathlib.Path(f"{project_path}/inputs/images")
+
+    # Run function to compile the index.csv
+    index_csv = compile_index_csv(images_load_path, DP_images_path, annotations, object)
+    index_csv.to_csv(index_save_path, index=False)
+    print("index.csv file saved!")
+
+    # Compile and save locations to DP project for specified object
+    locations_save_path = pathlib.Path(f"{project_path}/inputs/locations/")
+    print("Compiling locations!")
+    compile_training_locations(
+        index_save_path, segmentation_data_path, locations_save_path, object
+    )
+    print("Done compiling locations!")
+
+
+def rename_cyto_locations(cyto_locations_path: pathlib.Path):
+    """Rename the .csv files within directory from "Nuclei.csv" to "Cytoplasm.csv" to avoid confusion during downstream analysis.
```

Comment:
Can you add a sentence or two in the docstring about why this renaming is necessary

Reply:
You can even add `see README for more details` as well :)

Reply:
I added that last suggestion. Much easier and better than repeating myself.